int64 id
float64 app_x
float64 app_y
float64 app_z
string hair_col
string hair_len